src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/app/ApplicationModule.java

```diff
@@ -100,9 +128,10 @@ public void configure(Binder binder) {
     binder.bind(ContainerEnvConfig.class).in(Singleton.class);
 
     binder.bind(Database.class).to(PostgresDatabase.class).in(Singleton.class);
-    binder.bind(Worker.class).to(ThreadPoolWorker.class).in(Singleton.class);
-    binder.bind(Worker.class).annotatedWith(Names.named(Repo.PARAMETERISATION_WORKER_NAME))
-        .to(ThreadPoolWorker.class).in(Singleton.class);
+
+    binder.install(new WorkerModule());
```

Comment:
This is all very painful. Could you not just do:

binder.bind(Worker.class).to(ThreadPoolWorker.class).annotatedWith(Names.named("workerName")).to("Default worker").in(Singleton.class)

and  similar for the other one?

Reply:
The steps it took to get that... it was even worse!

I don't think that works, because what annotation would you put on the ThreadPoolWorker itself to receive that String? It does seems like there ought to be a simple way to do this (know which annotation you are being created with), but if there is a way to do it, I haven't been able to find it.

One other alternative to the PrivateModule is perhaps MultiBinder, but that causes it's own difficulties...

I think this is probably the best solution with Guice but I'd love something simpler.

Reply:
```binder.bind(Worker.class).to(ThreadPoolWorker.class).annotatedWith(Names.named("workerName")).to("Default worker").in(Singleton.class)``` isn't valid because after the `to`, `in` is the only thing left available in the chain.

I can see what you want to do with that (bind the worker name whilst binding that worker) but I can't see how to actually do that in Guice except with a PrivateModule.

I have gotten another technique with an AssistedInjection factory to work, but I think it is even less clear that the PrivateModule. Will open a parallel PR with it.

Reply:
Ugh. OK. This is much horribleness. Thank you for investigating and sorry for the waste of time. I'll take your privatemodule approach